csaps/_sspumv.py

```diff
@@ -335,4 +390,10 @@ def _make_spline(x, y, w, smooth, shape, normalizedsmooth):
         c_shape = (4, pcount - 1) + shape[1:]
         c = np.vstack((c1, c2, c3, c4)).reshape(c_shape)
 
-        return c, p
+        # As calculating the GCV is a relatively expensive operation, we store the degrees of freedom
+        # required for the GCV calculation
+        if calculate_degrees_of_freedom:
+            degrees_of_freedom = p * (la.inv(p * w + ((1-p) * 6 * qt.T @ la.inv(r) @ qt)) @ w).trace()
```

Comment:
Could you explain the formula? For example, if  smoothing parameter `p` is equal to 0.0 we get a linear regression. In this case `degrees_of_freedom` will be 0.0. Shouldn't that equal 2 for the linear regression case?

I understand that `degrees_of_freedom` may not be an integer for a smoothing approximating spline, but specifically for smooth==0.0 is not entirely clear to me. Also we can check it for smooth==1.0, when we get natural cubic spline without smoothing. In this case `degrees_of_freedom` should be the number of interior knots + 1, if I understand correctly. https://www.utstat.toronto.edu/reid/sta450/feb23.pdf